samza-core/src/main/java/org/apache/samza/clustermanager/ResourceRequestState.java

```diff
@@ -229,6 +230,25 @@ public int releaseExtraResources() {
     }
   }
 
+  /**
+   * Releases a single resource based on containerId.
+   * @param containerId Yarn container ID
+   */
+  public void releaseResource(String containerId) {
+    if (StringUtils.isEmpty(containerId)) {
+      log.warn("ContainerId is not specified");
```

Comment:
please add "ContainerId is not specified" 'in ClusterManager release resource'.
Without the context the message may be to vague.